drivers/sensor/grow_r502a/grow_r502a.h

```diff
@@ -0,0 +1,170 @@
+#define FPS_COMMAND_PACKET		0x1 /*Command packet*/
+#define FPS_DATA_PACKET			\
+	0x2 /*Data packet, must follow command packet or acknowledge packet*/
+#define FPS_ACK_PACKET			0x7 /*Acknowledge packet*/
+#define FPS_END_DATA_PACKET		0x8 /*End of data packet*/
+
+/*Instruction code's definition*/
+#define FPS_GENIMAGE			0x01 /*Collect finger image*/
+#define FPS_IMAGE2TZ			0x02 /*To generate character file from image*/
+#define FPS_MATCH			0x03 /*Carry out precise matching of two templates*/
+#define FPS_SEARCH			0x04 /*Search the finger library*/
+#define FPS_REGMODEL			0x05 /*To combine character files and generate template*/
+#define FPS_STORE			0x06 /*To store template*/
+#define FPS_LOAD			0x07 /*To read/load template*/
+#define FPS_UPCHAR			0x08 /*To upload template*/
+#define FPS_DOWNCHAR			0x09 /*To download template*/
+#define FPS_IMGUPLOAD			0x0A /*To upload image*/
+#define FPS_DELETE			0x0C /*To delete template*/
+#define FPS_EMPTYLIBRARY		0x0D /*To empty the library*/
+#define FPS_SETSYSPARAM			0x0E /*To set system parameter*/
+#define FPS_READSYSPARAM		0x0F /*To read system parameter*/
+#define FPS_SETPASSWORD			0x12 /*To set password*/
+#define FPS_VERIFYPASSWORD		0x13 /*To verify password*/
+#define FPS_GETRANDOM			0x14 /*To generate a random code*/
+#define FPS_TEMPLATECOUNT		0x1D /*To read finger template numbers*/
+#define FPS_READTEMPLATEINDEX		0x1F /*Read fingerprint template index table*/
+#define FPS_LED_CONFIG			0x35 /*Aura LED Config*/
+#define FPS_CHECKSENSOR			0x36 /*Check sensor*/
+#define FPS_SOFTRESET			0x3D /*Soft reset*/
+#define FPS_HANDSHAKE			0x40 /*Handshake*/
+#define FPS_BADPACKET			0xFE /* Bad packet was sent*/
+
+#define FPS_STARTCODE			0xEF01 /*Fixed value, High byte transferred first*/
+#define FPS_DEFAULT_PASSWORD		0x00000000
+#define FPS_DEFAULT_ADDRESS		0xFFFFFFFF
+#define FPS_DEFAULT_CAPACITY		200
+#define FPS_HANDSHAKE_BYTE		0x55
+
+#define FPS_HEADER_LEN			9
+#define FPS_CHECKSUM_LEN		2 /* Checksum length in uart packages*/
+#define FPS_BUF_SIZE			368 /* Maximum Characteristic value size*/
+
+#define DEFAULT_TIMEOUT			1000 /*UART reading timeout in ms*/
+#define FPS_LED_DELAY			100
+
+#define LED_CTRL_BREATHING		0x01
+#define LED_CTRL_FLASHING		0x02
+#define LED_CTRL_ON_ALWAYS		0x03
+#define LED_CTRL_OFF_ALWAYS		0x04
+#define LED_CTRL_ON_GRADUALLY		0x05
+#define LED_CTRL_OFF_GRADUALLY		0x06
+
+#define LED_SPEED_HALF			0x50
+#define LED_SPEED_FULL			0xFF
+
+#define LED_COLOR_RED                   0x01
+#define LED_COLOR_BLUE                  0x02
+#define LED_COLOR_PURPLE                0x03
+
+struct led_params {
+	uint8_t color_code;
+	uint8_t color;
+	uint8_t speed; /* Speed 0x00-0xff */
+	uint8_t cycle; /* Number of cycles | 0-infinite, 1-255 */
+};
+
+struct grow_r502a_data {
+	const struct device *dev;
+	char *uart_bus;
+
+	struct gpio_dt_spec vin_gpios;
+	struct gpio_dt_spec act_gpios;
+
+#ifdef CONFIG_GROW_R502A_TRIGGER
+	const struct device *uart_dev;
+	struct gpio_dt_spec  int_gpios;
```

Comment:
Please move all the devices and gpio_dt_spec values to a driver configuration structure to save RAM, since they can be const. Then you can remove the unncessary char* field.